nsqd/http.go

```diff
@@ -114,6 +114,58 @@ func (s *httpServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	s.router.ServeHTTP(w, req)
 }
 
+func (s *httpServer) isAuthEnabled(req *http.Request) bool {
+	if s.isTLS(req) == 1 {
+		return s.ctx.nsqd.IsAuthEnabledHTTPS()
+	}
+	return s.ctx.nsqd.IsAuthEnabledHTTP()
+}
+
+func (s *httpServer) isTLS(req *http.Request) int32 {
+	if req.TLS != nil {
+		return 1
+	}
+	return 0
+}
+
+func (s *httpServer) checkAuth(req *http.Request, cmd, topicName, channelName string) error {
+	if !s.isAuthEnabled(req) {
+		return nil
+	}
+
+	// TODO: Need to determine how to cache auth responses
+	authService := AuthService{
+		AuthParameters: AuthParameters{
+			AuthHTTPAddresses: s.ctx.nsqd.getOpts().AuthHTTPAddresses,
+			RemoteIP:          req.RemoteAddr,
+			TLS:               s.isTLS(req),
+		},
+	}
+
+	err := authService.Auth("TODO_SECRET")
+	if err != nil {
+		// we don't want to leak errors contacting the auth server to untrusted clients
+		s.ctx.nsqd.logf("ERROR: [%s] Auth Failed %s", s, err)
+		return http_api.Err{500, "E_AUTH_FAILED"}
+	}
+	// TODO: We'll need something like this if the secret isn't passed with the request
+	/*if !auth.HasAuthorizations() {
+		return http_api.Err{401, "E_AUTH_FIRST"}
+	}*/
```

Comment:
Reminder to check the Header or query param for the `secret` and return a 401 if not present.

Should a query param be supported or just a header? Any preference for naming?